### PR TITLE
UMD-4: add VOMS personality

### PR DESCRIPTION
**Pull request by jouvin**

### CODE REVIEW

File: personality/voms/config.pan [hunk 1]
```diff
@@ -0,0 +1,34 @@
+unique template personality/voms/config;
+
+include { 'personality/voms/variables' };
```
Comment (jrha) [inline personality/voms/config.pan]: Remove curlies from all the files in this PR please.

File: personality/voms/config.pan [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+include { 'personality/voms/mysql' };
+include { 'personality/voms/vos' };
+include { 'features/gip/base' };
+
+include { 'components/filecopy/config' };
+prefix '/software/components/filecopy/services';
+'{/etc/sysconfig/voms}'=
+  nlist(
```
Comment (jrha) [inline personality/voms/config.pan]: `dict`?

File: personality/voms/config.pan [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    'config', file_contents('personality/voms/files/voms.sysconfig'),
+    'owner', 'root:root',
+    'perms', '0644',
+);
+
+'{/etc/sysconfig/voms-admin}' =
+  nlist(
```
Comment (jrha) [inline personality/voms/config.pan]: `dict`?

File: personality/voms/config.pan [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+                     VOMS_TOMCAT_MX,
+                     VOMS_TOMCAT_MAXPERMSIZE
+                     ),
+    'owner', 'root:root',
+    'perms', '0644',
+);
+
+include { 'components/chkconfig/config' };
+
+ "/software/components/chkconfig/service/voms/on" = "";
```
Comment (jrha) [inline personality/voms/config.pan]: Use `prefix` for these too?

File: personality/voms/service.pan [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+variable VOMS_CONFIG_SITE ?= null;
+variable NODE_USE_RESOURCE_BDII = true;
+
+# Database server used by VOMS (default: local node)
+variable VOMS_DB_SERVER ?= FULL_HOSTNAME;
+
+
+# Include VOMS rpms.
+'/software/packages/{emi-voms-mysql}' ?= nlist();
```
Comment (jrha) [inline personality/voms/service.pan]: `pkg_repl`?

File: personality/voms/service.pan [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+# Add site specific configuration (mandatory)
+variable VOMS_CONFIG_SITE = if ( exists(VOMS_CONFIG_SITE) && is_defined(VOMS_CONFIG_SITE) ) {
```
Comment (jrha) [inline personality/voms/service.pan]: Would this accomplish the same thing?
```pan
variable VOMS_CONFIG_SITE ?= error("VOMS_CONFIG_SITE must be defined (VOMS server configuration)");
```

File: personality/voms/tomcat.pan [hunk 7]
```diff
@@ -0,0 +1,65 @@
+unique template personality/voms/tomcat;
+
+include { 'components/symlink/config' };
+prefix '/software/components/symlink';
+
+'links'= push(
+  nlist("name",    VOMS_TOMCAT_COMMON_LIB+"/xalan-j2-serializer.jar",
```
Comment (jrha) [inline personality/voms/tomcat.pan]: `dict`

File: personality/voms/tomcat.pan [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+    'perms', '0755',
+    'restart', '/var/run/quattor/certificate-copy.sh',
+);
+
+variable CONTENTS = VOMS_TOMCAT_USER + " soft nofile 65536\n";
+variable CONTENTS = CONTENTS + VOMS_TOMCAT_USER + " hard nofile 65536\n";
+
+'services'={
+  if ( VOMS_TOMCAT_MANAGE_LIMITS_FILE ) {
+    SELF[escape('/etc/security/limits.conf')]=
```
Comment (jrha) [inline personality/voms/tomcat.pan]: https://quattor-documentation.readthedocs.io/en/latest/components/interactivelimits/

File: personality/voms/vos.pan [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+      contents = contents + " --createdb --deploy-database --dbauser "+params['dbUser'];
+      contents = contents + " --dbapwd '"+params['dbPassword']+"' --dbusername "+params['dbUser'];
+      contents = contents + " --dbpassword '"+params['dbPassword']+"' --core-port "+params['port'];
+      contents = contents + " --dbname "+params['dbName'];
+      if ( exists(params['timeout']) ) {
+        contents = contents + ' --timeout '+params['timeout'];
+      };
+      contents = contents + ' --csrf-log-only ';
+      contents = contents + " --mail-from "+params['adminEmail']+" --smtp-host "+VOMS_ADMIN_SMTP_HOST+"\n";
+      contents = contents + "/usr/sbin/voms-db-util add-admin --vo "+vo+" --cert /etc/grid-security/hostcert.pem\n";
```
Comment (jrha) [inline personality/voms/vos.pan]: `join()` would be much, much, much nicer than this

File: personality/voms/vos.pan [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+    'perms', '0755',
+    'restart', if (VOMS_TOMCAT_RESTART_CONFIG) { '/var/run/quattor/create-voms.sh'; } else { null; }    
+);
+
+#/usr/sbin/voms-admin-configure install --dbtype mysql \ 
+#--vo test_vo_mysql --createdb --deploy-database \ 
+#--dbauser root --dbapwd pwd \ 
+#--dbusername voms_admin_20 --dbpassword pwd \ 
+#--port 54322 --mail-from ciccio@cnaf.infn.it \ 
+#–smtp-host iris.cnaf.infn.it
```
Comment (jrha) [inline personality/voms/vos.pan]: Remove these comments, or add some explanation as to why they are here.

File: personality/voms/config.pan [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+
+include { 'personality/voms/variables' };
+
+include { 'personality/voms/mysql' };
+include { 'personality/voms/vos' };
+include { 'features/gip/base' };
+
+include { 'components/filecopy/config' };
+prefix '/software/components/filecopy/services';
+'{/etc/sysconfig/voms}'=
```
Comment (jrha) [inline personality/voms/config.pan]: https://quattor-documentation.readthedocs.io/en/latest/components/sysconfig/

File: personality/voms/files/log4j-trustmanager.properties [hunk 12]
```diff
@@ -0,0 +1,7 @@
+log4j.logger.org.glite.security=INFO, fileout
```
Comment (jrha) [inline personality/voms/files/log4j-trustmanager.properties]: Pretty sure ncm-metaconfig can write `.properties` files, but this one doesn't seem to be used in this PR.

File: personality/voms/files/server.xml [hunk 13]
```diff
@@ -0,0 +1,21 @@
+<Server port="8005" shutdown="SHUTDOWN">
```
Comment (jrha) [inline personality/voms/files/server.xml]: This file also doesn't seem to be referenced anywhere, but also... ncm-metaconfig.

File: personality/voms/files/tomcat5.conf [hunk 14]
```diff
@@ -0,0 +1,12 @@
+CATALINA_HOME="%s"
```
Comment (jrha) [inline personality/voms/files/tomcat5.conf]: Definitely ncm-metaconfig, but this file also doesn't seem to be referenced anywhere in this PR.

File: personality/voms/files/voms-admin.sysconfig [hunk 15]
```diff
@@ -0,0 +1,19 @@
+#
```
Comment (jrha) [inline personality/voms/files/voms-admin.sysconfig]: ncm-sysconfig

File: personality/voms/files/voms.sysconfig [hunk 16]
```diff
@@ -0,0 +1,5 @@
+#
```
Comment (jrha) [inline personality/voms/files/voms.sysconfig]: ditto

**Comment by jrha** [comment]
@jouvin I've cleaned this up a bit, if my changes look okay to you we can squash them and merge.

**Comment by jouvin** [comment]
I'll try to have a look asap...

**Comment by jrha** [comment]
Thanks!

**Comment by jrha** [comment]
Apologies, I found a few more things to fix.

**Pull request merged**